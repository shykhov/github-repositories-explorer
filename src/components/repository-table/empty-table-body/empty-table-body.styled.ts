import styled from 'styled-components';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

export const StyledTableCell = styled(TableCell)`
  font-weight: bold !important;
  font-size: 18px !important;

  & > span {
    font-size: 40px;
  }
`;

export const StyledTableRow = styled(TableRow)`
  height: 384px;
`;
