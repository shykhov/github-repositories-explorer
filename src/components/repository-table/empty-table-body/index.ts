export * from './empty-table-body';
