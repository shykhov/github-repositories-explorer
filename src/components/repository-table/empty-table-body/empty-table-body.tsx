import React, { FC } from 'react';
import Link from '@material-ui/core/Link';

import { ContentRenderer } from '../../content-renderer';
import { AUTHOR_GITHUB_PROFILE_URL } from '../../../constants';
import { StyledTableRow, StyledTableCell } from './empty-table-body.styled';

export interface Props {
  called: boolean;
  error: boolean;
  hasNoQueryParameters: boolean;
  isEmptyRepositoriesList: boolean;
}

export const EmptyTableBody: FC<Props> = ({ called, error, hasNoQueryParameters, isEmptyRepositoriesList }) => (
  <StyledTableRow>
    <StyledTableCell key="empty-data" align="center" colSpan={3}>
      <ContentRenderer
        hasError={error}
        isEmpty={called && isEmptyRepositoriesList && !hasNoQueryParameters}
        emptyComponent={
          <>
            <span aria-label="emodji" role="img">
              🤷🏻‍♂️
            </span>{' '}
            <div>We couldn’t find any results. Let’s try removing some filters.</div>
          </>
        }
        errorComponent={
          <>
            <span aria-label="emodji" role="img">
              🙅🏻‍♂️
            </span>
            <div>
              Something went wrong. Please{' '}
              <Link href={AUTHOR_GITHUB_PROFILE_URL} target="_blank" rel="noopener noreferrer">
                contact the author
              </Link>{' '}
              of the project.
            </div>
          </>
        }
        contentComponent={
          <>
            <span aria-label="emodji" role="img">
              🕵🏻
            </span>{' '}
            <div>Enter repository name or choose github user to start our journey.</div>
          </>
        }
      />
    </StyledTableCell>
  </StyledTableRow>
);
