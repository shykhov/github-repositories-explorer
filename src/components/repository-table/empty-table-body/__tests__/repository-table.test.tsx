import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import { EmptyTableBody } from '../empty-table-body';

describe('<EmptyTableBody />', () => {
  let renderedComponent: ShallowWrapper;
  let propsMock;

  beforeEach(() => {
    propsMock = {
      error: true,
      called: true,
    };

    renderedComponent = shallow(<EmptyTableBody {...propsMock} />);
  });

  it('should render EmptyTableBody with error message', () => {
    expect(renderedComponent.dive()).toMatchSnapshot();
  });

  it('should render EmptyTableBody with empty message based on filters state', () => {
    renderedComponent.setProps({ called: true, error: false });

    expect(renderedComponent.dive()).toMatchSnapshot();
  });

  it('should render EmptyTableBody with empty message based on lack of filters', () => {
    renderedComponent.setProps({ called: false, error: false });

    expect(renderedComponent.dive()).toMatchSnapshot();
  });
});
