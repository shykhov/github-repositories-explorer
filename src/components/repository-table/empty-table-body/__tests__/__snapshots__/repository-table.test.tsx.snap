// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyTableBody /> should render EmptyTableBody with empty message based on filters state 1`] = `
.c0 {
  height: 384px;
}

<WithStyles(ForwardRef(TableRow))
  className="c0"
>
  <Styled(WithStyles(ForwardRef(TableCell)))
    align="center"
    colSpan={3}
    key="empty-data"
  >
    <ContentRenderer
      contentComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            ü§¨
          </span>
           
          <div>
            We couldn‚Äôt find any results. Let‚Äôs try removing some filters.
          </div>
        </React.Fragment>
      }
      emptyComponent={null}
      errorComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            üí©
          </span>
          <div>
            Something went wrong. Please
             
            <ForwardRef(WithStyles)
              href="https://github.com/shykhov"
              rel="noopener noreferrer"
              target="_blank"
            >
              contact the author
            </ForwardRef(WithStyles)>
             
            of the project.
          </div>
        </React.Fragment>
      }
      hasError={false}
      isEmpty={false}
      isLoading={false}
      loadingComponent={null}
    />
  </Styled(WithStyles(ForwardRef(TableCell)))>
</WithStyles(ForwardRef(TableRow))>
`;

exports[`<EmptyTableBody /> should render EmptyTableBody with empty message based on lack of filters 1`] = `
.c0 {
  height: 384px;
}

<WithStyles(ForwardRef(TableRow))
  className="c0"
>
  <Styled(WithStyles(ForwardRef(TableCell)))
    align="center"
    colSpan={3}
    key="empty-data"
  >
    <ContentRenderer
      contentComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            üßê
          </span>
           
          <div>
            Enter repository name or choose github user to start our journey.
          </div>
           
        </React.Fragment>
      }
      emptyComponent={null}
      errorComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            üí©
          </span>
          <div>
            Something went wrong. Please
             
            <ForwardRef(WithStyles)
              href="https://github.com/shykhov"
              rel="noopener noreferrer"
              target="_blank"
            >
              contact the author
            </ForwardRef(WithStyles)>
             
            of the project.
          </div>
        </React.Fragment>
      }
      hasError={false}
      isEmpty={false}
      isLoading={false}
      loadingComponent={null}
    />
  </Styled(WithStyles(ForwardRef(TableCell)))>
</WithStyles(ForwardRef(TableRow))>
`;

exports[`<EmptyTableBody /> should render EmptyTableBody with error message 1`] = `
.c0 {
  height: 384px;
}

<WithStyles(ForwardRef(TableRow))
  className="c0"
>
  <Styled(WithStyles(ForwardRef(TableCell)))
    align="center"
    colSpan={3}
    key="empty-data"
  >
    <ContentRenderer
      contentComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            ü§¨
          </span>
           
          <div>
            We couldn‚Äôt find any results. Let‚Äôs try removing some filters.
          </div>
        </React.Fragment>
      }
      emptyComponent={null}
      errorComponent={
        <React.Fragment>
          <span
            aria-label="emodji"
            role="img"
          >
            üí©
          </span>
          <div>
            Something went wrong. Please
             
            <ForwardRef(WithStyles)
              href="https://github.com/shykhov"
              rel="noopener noreferrer"
              target="_blank"
            >
              contact the author
            </ForwardRef(WithStyles)>
             
            of the project.
          </div>
        </React.Fragment>
      }
      hasError={true}
      isEmpty={false}
      isLoading={false}
      loadingComponent={null}
    />
  </Styled(WithStyles(ForwardRef(TableCell)))>
</WithStyles(ForwardRef(TableRow))>
`;
